uvicorn/server.py

```diff
@@ -54,12 +55,23 @@ def __init__(self, config: Config) -> None:
         self.should_exit = False
         self.force_exit = False
         self.last_notified = 0.0
+        self._delayed_signal: Optional[signal.Signals] = None
 
     def run(self, sockets: Optional[List[socket.socket]] = None) -> None:
         self.config.setup_event_loop()
-        return asyncio.run(self.serve(sockets=sockets))
+        try:
+            return asyncio.run(self.serve(sockets=sockets))
+        except KeyboardInterrupt:
+            pass
 
     async def serve(self, sockets: Optional[List[socket.socket]] = None) -> None:
```

Comment:
Can we use the `_capture_exit_signals` as decorator instead? It would generate less footprint.

```suggestion
    @capture_signals
    async def serve(self, sockets: Optional[List[socket.socket]] = None) -> None:
```
In the suggestion, `capture_signals` = `_capture_exit_signals`.

Reply:
It's doable but would create a similar/larger footprint at another place.

In principle it is possible to refactor `_capture_exit_signals` to work as a decorator as well: a `contextlib.contextmanager` decorated *function* can be used as a decorator. However, this does not work for methods out of the box; I would have to write a new `contextlib.contextmanager`.

Reply:
I've experimented a bit with this but ultimately decided to drop my sketch for a context manager method. Stacking a context manager on top of a descriptor (for the `self` binding) on top of `async def` is much more complexity than a regular `contextmanager`.

Let me know if you would prefer the decorator variant anyway.

Reply:
I'm thinking if we should do something like:

```python
async def serve(self, sockets):
    with self._capture_exit_signals():
        self._serve(sockets)
```
This is not a suggestion, it's a thought.

Reply:
Good suggestion. The diff is much more on point now.